broadcast/internal/composite/broadcaster.go

```diff
@@ -31,6 +34,120 @@ func NewBroadcaster(strategy Strategy, factories ...BroadcastFactory) broadcast.
 	}
 }
 
+func (c *compositeBroadcaster) GetPolicyQuote(ctx context.Context) (*broadcast.PolicyQuoteResponse, error){
+    executionFuncs := make([]executionFunc, len(c.broadcasters))
+    for i, broadcaster := range c.broadcasters {
+        executionFuncs[i] = func(ctx context.Context) (Result, error) {
+            return broadcaster.GetPolicyQuote(ctx)
+        }
+    }
+
+    result, err := c.strategy.Execute(ctx, executionFuncs)
+    if err != nil {
+        return nil, err
+    }
+
+    policyQuoteResponse, ok := result.(*broadcast.PolicyQuoteResponse)
+    if !ok {
+        return nil, fmt.Errorf("unexpected result type: %T", result)
+    }
+
+    return policyQuoteResponse, nil
+}
+
+func (c *compositeBroadcaster) GetFeeQuote(ctx context.Context) (*broadcast.FeeQuote, error){
+    executionFuncs := make([]executionFunc, len(c.broadcasters))
+    for i, broadcaster := range c.broadcasters {
+        executionFuncs[i] = func(ctx context.Context) (Result, error) {
+            return broadcaster.GetFeeQuote(ctx)
+        }
+    }
+
+    result, err := c.strategy.Execute(ctx, executionFuncs)
+    if err != nil {
+        return nil, err
+    }
+
+    feeQuote, ok := result.(*broadcast.FeeQuote)
+    if !ok {
+        return nil, fmt.Errorf("unexpected result type: %T", result)
+    }
+
+    return feeQuote, nil
+}
+
+func (c *compositeBroadcaster) GetBestQuote(ctx context.Context) (*broadcast.FeeQuote, error) {
```

Comment:
If we want to query all broadcaster one by one I would prefer alloc-free version like:

```go
var bestQuote *broadcast.FeeQuote = nil

for _, broadcaster := range c.broadcasters {
	feeQuote, err := broadcaster.GetFeeQuote(ctx)
	if err == nil {

		if bestQuote == nil {
			bestQuote = feeQuote
		} else {
			feePer1000Bytes := (1000 / feeQuote.MiningFee.Bytes) * feeQuote.MiningFee.Satoshis
			bestFeePer1000Bytes := (1000 / bestQuote.MiningFee.Bytes) * bestQuote.MiningFee.Satoshis
			
			if feePer1000Bytes < bestFeePer1000Bytes {
				bestQuote = feeQuote
			}
		}

	}
}


if bestQuote == nil {
	return nil, errors.New("failed to get reponse from any miner")
}

return best, nil

```

TBH I would prefer querying broadcasters in parallel

Reply:
Nice ;-)